src/gtk/srchctrl.cpp

```diff
@@ -0,0 +1,429 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/srchctrl.cpp
+// Purpose:     wxSearchCtrl implementation - native
+// Author:      Kettab Ali
+// Created:     2019-12-23
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#if wxUSE_SEARCHCTRL
+
+#include "wx/srchctrl.h"
+
+#ifndef WX_PRECOMP
+    #include "wx/menu.h"
+#endif //WX_PRECOMP
+
+#include "wx/utils.h"
+#include "wx/gtk/private.h"
+#include "wx/gtk/private/gtk3-compat.h"
+
+
+#if GTK_CHECK_VERSION(3,6,0)
+    // GtkSearchEntry is available only for GTK+ >= 3.6
+    #define wxHAS_GTK_SEARCH_ENTRY 1
+#else // GTK < 3.6
+    #define wxHAS_GTK_SEARCH_ENTRY 0
+#endif // GTK >= 3.6
+
```

Comment:
I think it would be nice to have a simple inline
```cpp
inline bool HasGtkSearchEntry()
{
#ifdef wxHAS_GTK_SEARCH_ENTRY
    return wx_is_at_least_gtk3(6);
#else 
    return false;
#endif
}
```
too, just for readability.

Also note that it's important to have the `#else` branch, otherwise the library compiled with GTK < 3.6 but using 3.6+ during run-time wouldn't behave correctly (admittedly, this should be a pretty rare case, but it still might conceivably happen).